libraries/AP_Soaring/AP_Soaring.h

```diff
@@ -0,0 +1,163 @@
+    unsigned long _thermal_start_time_us;
+
+    // store time cruise was entered for hysteresis
+    unsigned long _cruise_start_time_us;
+
+    // store time of last update
+    unsigned long _prev_update_time;
+
+    // store time of last update
+    unsigned long _prev_vario_update_time;
+
+    float _vario_reading;
+    float _filtered_vario_reading;
+    float _last_alt;
+    float _alt;
+    float _last_aspd;
+    float _last_roll;
+    float _last_total_E;
+    bool _new_data;
+    float _loiter_rad;
+    bool _throttle_suppressed;
+    
+    float _aspd_filt;
+    uint8_t _msgid;
+    uint8_t _msgid2;
+    float correct_netto_rate(float climb_rate, float phi, float aspd);
+    float McCready(float alt);
+    void get_wind_corrected_drift(const Location *current_loc, const Vector3f *wind, float *wind_drift_x, float *wind_drift_y, float *dx, float *dy);
+    void get_altitude_wrt_home(float *alt);
+  
+protected:
+    AP_Int8 soar_active;
+    AP_Float thermal_vspeed;
+    AP_Float thermal_q1;
+    AP_Float thermal_q2;
+    AP_Float thermal_r;
+    AP_Float thermal_distance_ahead;
+    AP_Int16 min_thermal_s;
+    AP_Int16 min_cruise_s;
+    AP_Float polar_CD0;
+    AP_Float polar_B;
+    AP_Float polar_K;
+    AP_Float alt_max;
+    AP_Float alt_min;
+    AP_Float alt_cutoff;
+        
+public:
+    SoaringController(AP_AHRS &ahrs, AP_SpdHgtControl *&spdHgt, const AP_Vehicle::FixedWing &parms, uint8_t msgid, uint8_t msgid2) :
+        _ahrs(ahrs),
+        _spdHgt(spdHgt),
+        _aparm(parms),
+        _thermal_start_time_us(0),
+        _cruise_start_time_us(0),
+        _prev_update_time(0),
+        _prev_vario_update_time(0),
+        _vario_reading(0.0f),
+        _filtered_vario_reading(0.0f),
+        _last_alt(0.0f),
+        _alt(0.0f),
+        _last_aspd(0.0f),
+        _last_roll(0.0f),
+        _last_total_E(0.0f),
+        _new_data(false),
+        _loiter_rad(parms.loiter_radius),
+        _throttle_suppressed(true),
+        _aspd_filt(0.0f),
+        _msgid(msgid),
+        _msgid2(msgid2)
+    {
+        AP_Param::setup_object_defaults(this, var_info);
+        ahrs.get_position(_prev_update_location);
+        ahrs.get_position(_prev_vario_update_location);  
+    }
+    
+    float _displayed_vario_reading;  
+    // this supports the TECS_* user settable parameters
+    static const struct AP_Param::GroupInfo var_info[];
+    void get_target(Location & wp);
+    bool suppress_throttle();
+    void log_data();
+    bool check_thermal_criteria();
+    bool check_cruise_criteria();
+    bool check_init_thermal_criteria();
+    void init_thermalling();
+    void init_cruising();
+    void update_thermalling();
+    void update_cruising();
+    bool is_active();
+    bool get_throttle_suppressed() { return _throttle_suppressed; }
+    void set_throttle_suppressed(bool suppressed) { _throttle_suppressed = suppressed; }
+    
+    // Soaring log structure
+    struct PACKED log_Thermal_Tuning {
```

Comment:
you can get rid of all of these log bits once you move to Log_Write() as above (see AP_TECS example)